lib/elixir/src/elixir_module.erl

```diff
@@ -54,6 +54,8 @@ next_counter(Module) ->
 
 %% Compilation hook
 
+compile(Module, _Block, _Vars, #{line := Line, file := File}) when Module == nil; is_boolean(Module) ->
+  elixir_errors:form_error([{line, Line}], File, ?MODULE, {invalid_module, Module});
```

Comment:
Beautiful!

Reply:
@josevalim would you prefer it there, or in `kernel.ex` as @whatyouhide had suggested?

Reply:
I prefer here, you win @rodrigues! Sorry @whatyouhide. :P

Seriously though, where you put also forbids `Module.create(true, ...)`. Where @whatyouhide suggested, it wouldn't.

Reply:
Ah, didn't think of `Module.create/3`, good call.